### ISSUE TITLE
Bug: Filtered Vehicle Positions GTFS-RT Bigquery returns type error

### ISSUE THREAD

**Issue by edasmalchi** [cal-itp/data-infra#811]
**Describe the bug**

When I try to query our vehicle positions data for a specific operator and timeframe, I get a column type mismatch error that doesn't seem related to my query:

```
Error while reading table: cal-itp-data-infra.gtfs_rt.vehicle_positions, error message: Parquet column 'vehicle_position_latitude' has type INT32 which does not match the target cpp_type DOUBLE. 
```

**To Reproduce**
On BigQuery:

```
SELECT *
FROM `cal-itp-data-infra.gtfs_rt.vehicle_positions`
WHERE calitp_itp_id = '203' --Modesto Area Express
    AND CAST(header_timestamp AS INT64) > 1637222400 --2021, 11, 18 (Thursday)
    AND CAST(header_timestamp AS INT64) < 1637654400 --2021, 11, 23 (Tuesday)
```

**Expected behavior**
Query returning appropriately filtered vehicle positions data

**Additional context**
Interestingly, adding a `LIMIT 100000` at the end of the query returns data. Perhaps this has something to do with exceeding the total number of existing records, but I want to make sure I'm getting every record available for the operator/timeframe.

**Comment by machow** [comment]
Ah, this is because the data was loaded to parquet without using a pre-defined schema on this line: https://github.com/cal-itp/data-infra/blob/main/airflow/dags/rt_loader/parse_rt_vehicle_positions.py#L174

In this case, it is inferring the schema from the data. When lattitude just so happens to not have any decimals (not sure why that would happen), it causes this error.

### AFAICT to fix:

* create explicit schema in the linked task
* re-run task across history

Note that it appears this task sometimes fails, due to an issue with gcsfs periodically failing on requests. We've used strategies like enforcing a retry in the past, e.g. in the fs.open used with `parse_pb` [on this line](https://github.com/cal-itp/data-infra/blob/main/airflow/dags/rt_loader/parse_rt_vehicle_positions.py#L40). Alternatively, could use `fs.put` with a list of files to save all at once, and then open them.

**Comment by machow** [comment]
@edasmalchi in the meantime, you should be able to query individual parquet files using this query

```SQL
# get all vehicle positions on July 12th, for the feed with itp id 75, and url number 0
SELECT *
FROM `cal-itp-data-infra.gtfs_rt.vehicle_positions`
WHERE _FILE_NAME="gs://gtfs-data/rt-processed/vehicle_positions/vp_2021-07-12_75_0.parquet"
```

You can list out or search file names using this query..!:

```SQL
SELECT DISTINCT _FILE_NAME AS fn FROM `cal-itp-data-infra.gtfs_rt.vehicle_positions`
```

**Comment by edasmalchi** [comment]
thanks Michael!

**Comment by holly-g** [comment]
Closed by #822
